src/java/org/apache/cassandra/index/sai/disk/OnDiskKeyProducer.java

```diff
@@ -75,25 +71,12 @@ private long getKeyOffset(long segmentRowId)
             return NO_OFFSET;
         }
 
-        // Due to ZCS, index files may still contain partition offsets that are not part of partial SSTable.
```

Comment:
Are the changes in this PR removing the possibility of this happening, or are we just removing an unnecessary check here?

Reply:
it's dead code. OSS doesn't support partial-sstable zero-copy-streaming..